Change: correctly propagate ElasticSearch errors to the UI

adamlwgriffiths opened the pull request:
All erronous ES queries have been displaying an error stating the JSON could not be decoded, regardless of the actual error (HTTP, syntax, auth, etc).

This modification ensures any request error is propagated back.
It seems that the main issue was requests errors being non-json-serializable due to the usage of both " and ' characters.
The abstraction of the `_get_mapping` function also made it difficult to get the root error instead of just the exception, this has been modified to be similar to `run_query` and now returns a tuple of `(mappings, error)`.

arikfr commented:
Thank you for doing some house keeping there as well! The ElasticSearch data source definitely needs some more love.

adamlwgriffiths commented:
Just noticed that I've mixed logging and logger usage (always do that =\). I'll fix this tomorrow and update the PR. 

> On 17 Nov 2016, at 7:23 pm, Arik Fraimovich <notifications@github.com> wrote:
> 
> Thank you for doing some house keeping there as well! The ElasticSearch data source definitely needs some more love.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

adamlwgriffiths commented:
Ok, should be good to go.
Changed incorrect usage of `logging` to `logger`.
Moved some of the syntax changes from the error commit back to the PEP-8 commit.
